From 908eee07cd6b27fd69ac7dc969950e6c5b36c14d Mon Sep 17 00:00:00 2001
From: Gyorgy Szing <Gyorgy.Szing@arm.com>
Date: Mon, 5 Jul 2021 00:33:59 +0000
Subject: [PATCH 1/1] Add install definition
+
+Add install() calls to define stable way to access build artifacts.

Signed-off-by: Gyorgy Szing <Gyorgy.Szing@arm.com>
---
 CMakeLists.txt | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0b01d8b..343b325 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -31,4 +31,17 @@ else()

 endif()

+file(GLOB_RECURSE _t_cose_headers LIST_DIRECTORIES false "${CMAKE_CURRENT_SOURCE_DIR}/inc/*.h")
+set_property(TARGET t_cose APPEND PROPERTY PUBLIC_HEADER ${_t_cose_headers})
+
+install(
+	TARGETS
+		t_cose
+	ARCHIVE	DESTINATION
+		lib
+	PUBLIC_HEADER DESTINATION
+		include/t_cose
+	COMPONENT
+		t_cose
+)

--
2.17.1

